src/shared/lib/CancerHotspotsUtils.ts

```diff
@@ -0,0 +1,66 @@
+import * as _ from "lodash";
+import MobxPromise from "mobxpromise";
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import {
+    default as GenomeNexusAPIInternal, AggregatedHotspots, GenomicLocation, Hotspot
+} from "shared/api/generated/GenomeNexusAPIInternal";
+import genomeNexusInternalClient from "shared/api/genomeNexusInternalClientInstance";
+import {concatMutationData} from "./StoreUtils";
+import {extractGenomicLocation} from "./MutationUtils";
+import {IHotspotIndex} from "shared/model/CancerHotspots";
+
+export async function fetchHotspotsData(mutationData: MobxPromise<Mutation[]>,
+                                        uncalledMutationData?: MobxPromise<Mutation[]>,
+                                        client: GenomeNexusAPIInternal = genomeNexusInternalClient)
+{
+    const mutationDataResult = concatMutationData(mutationData, uncalledMutationData);
+
+    if (mutationDataResult.length === 0) {
+        return [];
+    }
+
+    const genomicLocations: GenomicLocation[] = _.uniq(_.map(mutationDataResult, extractGenomicLocation));
+
+    return await client.fetchHotspotAnnotationByGenomicLocationPOST({genomicLocations: genomicLocations});
+}
+
+export function indexHotspotsData(hotspotData: MobxPromise<AggregatedHotspots[]>): IHotspotIndex|undefined
+{
+    if (hotspotData.result) {
+        return indexHotspots(hotspotData.result);
+    }
+    else {
+        return undefined;
+    }
+}
+
+export function indexHotspots(hotspots: AggregatedHotspots[]): IHotspotIndex
+{
+    const index:IHotspotIndex = {};
+
+    hotspots.forEach((aggregatedHotspots: AggregatedHotspots) => {
+        index[genomicLocationString(aggregatedHotspots.genomicLocation)] = aggregatedHotspots;
+    });
+
+    return index;
+}
+
+export function genomicLocationString(genomicLocation: GenomicLocation)
+{
+    return `${genomicLocation.chromosome},${genomicLocation.start},${genomicLocation.end},${genomicLocation.referenceAllele},${genomicLocation.variantAllele}`;
+}
+
+export function isHotspot(mutation: Mutation, index: IHotspotIndex, filter?: (hotspot: Hotspot) => boolean): boolean
+{
+    let isHotspot = false;
```

Comment:
shadowing is sort of confusing to read IMO - better to have different name for variable